Detectors/FIT/T0/simulation/src/Detector.cxx

```diff
@@ -217,90 +218,172 @@ void Detector::ConstructOpGeometry()
+      z += ptopref[2] + 2. * pmcptopglass[2] + preg[2];
       ins->AddNode(cat, ntops, new TGeoTranslation(xin, yin, z));
+      cat->Print();
+      printf(" GEOGEO  CATHOD x=%f , y= %f z= %f num  %i\n", xin, yin, z, ntops);
     }
   }
-  // Al top
+  //Al top
   z = -pinstart[2] + pal[2];
   ins->AddNode(altop, 1, new TGeoTranslation(0, 0, z));
 
   // MCP
-  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + 2 * preg[2] + pmcp[2];
-  //   z=-pinstart[2] + 2*ptopref[2] + preg[2];
+  TVirtualMC::GetMC()->Gsvolu("0MTO", "BOX", getMediumID(kOpGlass), pmcptopglass, 3); //Op  Glass
+  TGeoVolume* mcptop = gGeoManager->GetVolume("0MTO");
+  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + pmcptopglass[2];
+  ins->AddNode(mcptop, 1, new TGeoTranslation(0, 0, z));
+
+  TVirtualMC::GetMC()->Gsvolu("0MCP", "BOX", getMediumID(kAir), pmcp, 3); //glass
+  TGeoVolume* mcp = gGeoManager->GetVolume("0MCP");
+  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + 2 * pmcptopglass[2] + 2 * preg[2] + pmcp[2];
   ins->AddNode(mcp, 1, new TGeoTranslation(0, 0, z));
+  TVirtualMC::GetMC()->Gsvolu("0MIN", "BOX", getMediumID(kGlass), pmcpinner, 3); //glass
+  TGeoVolume* mcpinner = gGeoManager->GetVolume("0MIN");
+  mcp->AddNode(mcpinner, 1, new TGeoTranslation(0, 0, 0));
+
+  TVirtualMC::GetMC()->Gsvolu("0MSI", "BOX", getMediumID(kGlass), pmcpside, 3); //glass
+  TGeoVolume* mcpside = gGeoManager->GetVolume("0MSI");
+  x = -pmcp[0] + pmcpside[0];
+  y = -pmcp[1] + pmcpside[1];
+  mcp->AddNode(mcpside, 1, new TGeoTranslation(x, y, 0));
+  x = pmcp[0] - pmcpside[0];
+  y = pmcp[1] - pmcpside[1];
+  mcp->AddNode(mcpside, 2, new TGeoTranslation(x, y, 0));
+  x = -pmcp[1] + pmcpside[1];
+  y = -pmcp[0] + pmcpside[0];
+  mcp->AddNode(mcpside, 3, new TGeoCombiTrans(x, y, 0, new TGeoRotation("R2", 90, 0, 0)));
+  x = pmcp[1] - pmcpside[1];
+  y = pmcp[0] - pmcpside[0];
+  mcp->AddNode(mcpside, 4, new TGeoCombiTrans(x, y, 0, new TGeoRotation("R2", 90, 0, 0)));
+
+  TVirtualMC::GetMC()->Gsvolu("0MBA", "BOX", getMediumID(kCeramic), pmcpbase, 3); //glass
+  TGeoVolume* mcpbase = gGeoManager->GetVolume("0MBA");
+  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + pmcptopglass[2] + 2 * pmcp[2] + pmcpbase[2];
+  ins->AddNode(mcpbase, 1, new TGeoTranslation(0, 0, z));
+
+  // Al Housing for Support Structure
+  //  ins->AddNode(alsup,1);
 }
 
 Bool_t Detector::ProcessHits(FairVolume* v)
 {
+  TVirtualMCStack* stack = fMC->GetStack();
   Int_t quadrant, mcp;
   if (fMC->IsTrackEntering()) {
     float x, y, z;
     fMC->TrackPosition(x, y, z);
     fMC->CurrentVolID(quadrant);
     fMC->CurrentVolOffID(1, mcp);
     float time = fMC->TrackTime() * 1.0e9; //time from seconds to ns
-    int trackID = fMC->GetStack()->GetCurrentTrackNumber();
+    int trackID = stack->GetCurrentTrackNumber();
     int detID = 4 * mcp + quadrant - 1;
     float etot = fMC->Etot();
     int iPart = fMC->TrackPid();
     float enDep = fMC->Edep();
+    Int_t parentID = stack->GetCurrentTrack()->GetMother(0);
```

Comment:
Should not obtaining the parent ID moved into the ``if (cherenkov)`` where it is used.
Perhaps stupid questions, but (1) why do you need to register charged hits? (2) what about non-cherenkov photons, are EM showers really negligible?

Reply:
Hello
here are my answers:
1) registration of charged hits was temporary solution to check "true" multiplicity without additional uncertainty of photon generation and compare with AliRoot results. It will be removed when detector response will implemented and checked.
2)non cherenkov photons are not registered by MCP-PMT. I can draw quantum efficiency of photocathode as a function of photon energy if you
 want. 
Best regards
Alla

Reply:
ok for non-cerenkov photons. Still, I would move obtaining parent ID to the scope it is used.